module/User/src/Authentication/AuthenticationService.php

```diff
@@ -59,13 +58,14 @@ public function getAdapter()
      *
      * @return self Provides a fluent interface
      */
-    public function setAdapter(AdapterInterface $adapter)
+    public function setAdapter(AdapterInterface $adapter): static
```

Comment:
Is it really static what is returned? It seems more like an instance.

Reply:
In some locations we require `array`s, however, Laminas always returns `Parameters` (array-like). In https://github.com/GEWIS/gewisweb/pull/1158/commits/57c6a945f17c78f102a68b31d9a3c3b8b2bf8ffe I first made this change and while a bit vague we discussed it would be better to keep it `Parameters` in the controllers and handle any conversions in the services.

If needed I can always change all occurrences of this.

Reply:
Alright, noted.

Reply:
But this does not relate to the static/instance comment.

Reply:
@Koen1999 I cannot read, but `static` is a special class name that refers to the class a method was called on. In this specific instance (and the one below) we do `return $this;` which makes it a "fluent" method. See [PHP RFC: Static return type](https://wiki.php.net/rfc/static_return_type) for more information (and why `return $this;` is of return type `static`).